site/src/components/layout.md

```diff
@@ -0,0 +1,148 @@
+template: standard
+classType: Component
+classPrefixes:
+  - layout
+hero: true
+---
+
+# Layout
+
+The layout component provides a flexible and efficient way to structure the major containers of web pages.
+
+## Try layouts here
+
+This page is utilizing the holy grail layout, but you can use the buttons below to toggle layouts and test them out by resizing the screen. Try the "Toggle container labels" button below to see the css classes on each of the containers inside `.layout`.
+
+<div class="buttons buttons-addons margin-top-sm display-inline-flex">
+  <button class="button" data-set-layout="layout-holy-grail">Holy grail</button>
+  <button class="button" data-set-layout="layout-single">Single</button>
+</div>
+<button class="button margin-top-sm" data-toggle-debug>Toggle container labels</button>
+
+## Important concepts
+
+Thankfully, no one needs to be expert to use the `.layout` element. But it does help to understand a few important concepts before beginning.
+
+### Markup structure
+
+The `.layout` class is designed to be place on the `<html>` element itself, while the `layout-body` element is designed to be place on the `<body>` element. All other element are children of `layout-body`. Specific layouts are applied by adding a class with the format `layout-{nameoflayout}` to the `html.layout` element.
+
+Nearly all layouts have the same eight elements, and the `layout-body-hero` element is always optional. While the `layout`, `layout-body`, `layout-body-header`, `layout-body-footer`, and `layout-body-main` elements are always required. The behavior of `layout-body-menu` and `layout-body-aside` depend on the layout.
+
+The basic boilerplate for a layout is as follows:
+
+```html-no-example
+<html class="layout layout-{nameoflayout}">
+	<body class="layout-body">
+		<header style="layout-body-header">Header</header>
+		<div class="layout-body-hero">Hero</div>
+		<nav class="layout-body-nav">Menu</nav>
+		<main class="layout-body-main">Main</main>
+		<aside class="layout-body-aside">Aside</aside>
+		<footer class="layout-body-footer">Footer</footer>
+	</body>
+</html>
+```
+
+### Apply your own spacing
+
+The `$layout-gap` token is used to generate two separate classes that can be used to apply spacing to containers _within_ each of the major layout areas. Spacing should _not_ be applied to the `layout-body-*` containers themselves, but rather to elements within them - most often a single container that is a direct child of the `layout-body-*` element.
+
+| Class name        | Use case                             | Examples of use case                                                                  |
+| ----------------- | ------------------------------------ | ------------------------------------------------------------------------------------- |
+| `.layout-padding` | Layout areas that are not full width | The menu and aside areas. The main area when no in a single layout                    |
```

Comment:
`The main area when no in a single layout` -  is `no` supposed to be `not`? When not using the single layout?

Reply:
this should be fixed up.